File: pkg/spacetime/const.go
package spacetime

const LabelMaxLength = 128
const TextMaxLength = 2048
const TitleMaxLength = 256
const TagMaxLength = 128
const NakedTextMaxDeltasSoft = TextMaxLength * 3
const NakedTextMaxDeltas = TextMaxLength * 4

const SpaceTypeUser = "user"
const SpaceTypeBranch = "branch"
const SpaceTypeText = "text"
const SpaceTypeTag = "tag"
const SpaceTypeLink = "link"

// const SpaceTypeStream = "stream-of-consciousness"
// const SpaceTypeJson = "json-attribute"

func IsValidSpaceType(spaceType string) bool {
	switch spaceType {

	case SpaceTypeBranch,
		SpaceTypeText,
		SpaceTypeLink,
		SpaceTypeTag:
		return true

	default:
		// Not yet implemented
		return false
	}
}

func IsValidRootSpaceType(spaceType string) bool {
	switch spaceType {

	case SpaceTypeBranch:
		return true

	default:
		return false
	}
}

func IsAuthorContextSpace(spaceType string) bool {
	switch spaceType {

	case SpaceTypeUser, SpaceTypeText:
		return true

	default:
		return false
	}
}

File: src/const.js
export const SPACE_TYPES = {
	USER: 'user',
	BRANCH: 'branch',
	TEXT: 'text',
	LINK: 'link',
	TAG: 'tag',
	STREAM_OC: 'stream-of-consciousness',
	JSON_AR: 'json-attribute',
};

export const SPACE_TYPE_ICONS = {
	[SPACE_TYPES.USER]: 'person',
	[SPACE_TYPES.BRANCH]: 'fork_right',
	[SPACE_TYPES.TEXT]: 'description',
	[SPACE_TYPES.LINK]: 'link',
	[SPACE_TYPES.TAG]: 'label',
	[SPACE_TYPES.STREAM_OC]: 'stream',
	[SPACE_TYPES.JSON_AR]: 'code',
};

export const USER_CONTEXT_TYPES = [
	SPACE_TYPES.USER,
	SPACE_TYPES.TEXT,
];
